compiler/ccgtypes.nim

```diff
@@ -798,36 +798,59 @@ proc fillObjectFields*(m: BModule; typ: PType) =
 
 proc mangleDynLibProc(sym: PSym): Rope
 
-proc getRecordDescAux(m: BModule; typ: PType, name, baseType: Rope,
-                   check: var IntSet, hasField:var bool): Rope =
-  result = ""
-  if typ.kind == tyObject:
-    if typ.baseClass == nil:
-      if lacksMTypeField(typ):
-        appcg(m, result, " {$n", [])
+when defined(nimUseCBuilder):
```

Comment:
NO! Just make it use CBuilder's API. No `when` statements allowed.

Reply:
Yeah, I added that because it wasn't finished. It needs a refactor before using cbuilder. I will push easy changes for tuple first and refactor the code later

Reply:
Alright but the idea is that every little step keeps the bootstrapping process until `cbuilder` starts to do something interesting.